packages/formation-react/src/components/DownloadLink/DownloadLink.unit.spec.jsx

```diff
@@ -0,0 +1,348 @@
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+
+import DownloadLink, { FILE_TYPES, getExtGroup } from './DownloadLink';
+
+import {
+  renderFileSize,
+  processFileName,
+  getFileExtensionFromUrl,
+} from '../../helpers/link-utils';
+
+const testData = {
+  jpg: {
+    href: 'https://www.example.com/test.jpg',
+    title: 'Test image',
+    group: 'graphic',
+    type: 'image/jpeg',
+    size: '400b',
+    result: 'Download Test image JPG (400B)',
+  },
+  pdf: {
+    href: 'https://www.foo.bar.com/my.folder/file.pdf?value=3.4',
+    title: 'A much needed form',
+    group: 'Portable Document Format',
+    type: 'application/pdf',
+    size: '1.67mb',
+    result: 'Download A much needed form PDF (1.67MB)',
+  },
+  txt: {
+    href: 'https://www.test.com/file.txt#anchor-link',
+    title: 'Some random text file',
+    group: 'text file',
+    type: 'text/plain',
+    size: '321kb',
+    result: 'Download Some random text file TXT (321KB)',
+  },
+  docx: {
+    href: 'https://docs.com/mydoc.docx#table-1.1',
+    title: 'MS is the best',
+    group: 'Word document',
+    type:
+      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    size: '.03GB',
+    result: 'Download MS is the best DOCX (0.03GB)',
+  },
+};
+
+const testSizes = [
+  ['400b', '(400<abbr title="Bytes">B</abbr>)'],
+  ['1.67mb', '(1.67<abbr title="Megabytes">MB</abbr>)'],
+  ['321kb', '(321<abbr title="Kilobytes">KB</abbr>)'],
+  ['.03Gb', '(0.03<abbr title="Gigabytes">GB</abbr>)'],
+  [0, '(0<abbr title="Bytes">B</abbr>)'],
+  [199, '(199<abbr title="Bytes">B</abbr>)'],
+  [1000, '(1<abbr title="Kilobytes">KB</abbr>)'],
+  [1170, '(1.2<abbr title="Kilobytes">KB</abbr>)'],
+  [12345, '(12.3<abbr title="Kilobytes">KB</abbr>)'],
+  [3.2e6, '(3.2<abbr title="Megabytes">MB</abbr>)'],
+  [4.57e9, '(4.6<abbr title="Gigabytes">GB</abbr>)'],
+];
+
+const testIECSizes = [
+  [0, '(0<abbr title="Bytes">B</abbr>)'],
+  [1000, '(1000<abbr title="Bytes">B</abbr>)'],
+  [1050, '(1<abbr title="Kibibytes">KiB</abbr>)'],
+  [1170, '(1.1<abbr title="Kibibytes">KiB</abbr>)'],
+  [12345, '(12.1<abbr title="Kibibytes">KiB</abbr>)'],
+  [3.2e6, '(3.1<abbr title="Mebibytes">MiB</abbr>)'],
+  [4.57e9, '(4.3<abbr title="Gibibytes">GiB</abbr>)'],
+];
+
+describe('Widget <DownloadLink />', () => {
+  describe('utility functions', () => {
```

Comment:
Can you move these tests to a different file? I'd also like to see tests clearly articulate the edge cases for each utility.